pynequa/models.py

```diff
@@ -1,13 +1,48 @@
 from typing import Dict, List, Optional
-
-
-class TreeParams:
+from abc import abstractmethod
+from dataclasses import dataclass, field
+import logging
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+class SinequaBaseModel:
+    """
+    Abstract base class for all Sinequa models.
+    """
+
+    @abstractmethod
+    def generate_payload(self, **kwargs) -> Dict:
+        """
+        This is abstract method for SinequaBaseModel.
+        Every child class should implement this method.
+        """
+        raise NotImplementedError()
+
+
+@dataclass
+class TreeParams(SinequaBaseModel):
```

Comment:
Isn't the base model and Params/dataclass different? What's the difference?

Reply:
I think the naming might have confused you which I just realized as well. `SinequaBaseModel` is more of a base class that will have methods that other `Params` class will implement for now as there aren't any shared attributes which could be used across all.